### ISSUE TITLE
Docker image no longer 'serve's 

### ISSUE THREAD

**Issue by eddelbuettel** [squidfunk/mkdocs-material#2328]
- [x] I've read the [contribution guidelines][1] and agree with them

__I've found a bug and checked that ...__

- [x] ... the problem doesn't occur with the default MkDocs template
- [x] ... the problem is not in any of my customizations (CSS, JS, template)
- [x] ... the documentation does not mention anything about my problem
- [x] ... there are no open or closed issues that are related to my problem

## Description

`serve` no longer creates a browseable instance

This was verified on two dinstinct machines (though both are Ubuntu 20.10, current packages).  Docker images for mkdocs-material were freshly pulled.  In one case a new site was create using `new` followed by an attempt to `serve`.  In any case 127.0.0.1:8000 did not respond to the browser.  Difference ports, different base IPs besides localhost, ... made no difference.

This could well be a `mkdocs` issue but I see no mention of this in the `mkdocs-material` documentation, and it is a shortcoming as the live preview greatly helps in using the (wonderful) `mkdocs-material`.

### Expected behavior

A browsable site based on the `mkdocs.yaml` content etc.

### Actual behavior

"This site can't be reached" and "The webpage at http://127.0.0.1:8000/ might be temporarily down."

### Steps to reproduce the bug

```sh
docker run --rm -ti -v $PWD:/docs -p 8000:8000 squidfunk/mkdocs material new foo
cd foo
docker run --rm -ti -v $PWD:/docs -p 8000:8000 squidfunk/mkdocs material serve
```

### Package versions

Container with tag `latest` and image id `1787c2608e7d` listed as created 8 days ago.

### Project configuration

See above. Made by mkdocs

### System information

* OS: Ubuntu 20.10
* Browser: Chrome  88.0.4324.182 (Official Build) (64-bit)

  [1]: https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md

**Comment by squidfunk** [comment]
I just tested this and it works if you use:

```
docker run --rm -ti -v $PWD:/docs -p 8000:8000 squidfunk/mkdocs-material
```

`serve` is the default command. If you specify it explicitly, you need to add the `--dev-addr` flag, because MkDocs must bind on all interfaces to expose the dev server from guest to host:

https://github.com/squidfunk/mkdocs-material/blob/f27eb1ee07d5145ea2242013dffa2c3a6e1a6533/Dockerfile#L72

**Comment by eddelbuettel** [comment]
Concur, works here too! Thanks for clarifying!   

I was stumped earlier but was quite sure I had not been bitten before by this (I only dive into mkdocs-material every few weeks). I have another wish-list issue for you "eventually" but isn't quite ripe.  I am using this with R packages of mine and we often have `pandoc`-generated markdown.  I once tried 'wrapping' this inside mkdocs-material and failed over some css-related issues.  Don't quite have a minimal issue yet...

**Comment by squidfunk** [comment]
@eddelbuettel great! Regarding pandoc - we have to check how the Markup looks if it makes sense (or is even possible) to add the relevant styles, but I'll be curious to learn about the problems you're experiencing.

**Comment by eddelbuettel** [comment]
Sure thing. So in a nutshell "most" R packages on CRAN rely on an existing tool to generate per-package websites. But it is rigid in what it wants, and I happen to follow coding conventions it does not support (nothing major, but I sitll use GNU-style ChangeLog files, and use NEWS files in the Rd markup for R's help file) plus it prefers html docs and I had more pdf ones. Long story short I realized I could easily create and fill ChangeLog/ and News/ and man/ with minimal pre-processng in scripts (sometimes relying on `pandoc`), and have pdf vignettes in frames. I have a few mkdocs-material sites now:  [anytime](https://eddelbuettel.github.io/anytime/), [digest](https://eddelbuettel.github.io/digest/), [drat](https://eddelbuettel.github.io/drat/), [littler](https://eddelbuettel.github.io/littler/), [nanotime](https://eddelbuettel.github.io/nanotime/), [pkgKitten](https://eddelbuettel.github.io/pkgkitten/), [RcppSpdlog](https://eddelbuettel.github.io/rcppspdlog/).  And I am now forgetting which one it was but one gave me fits on the (R)md content when processed by mkdocs-material.  I'll find it one of these days and report back -- very glad to know you're not opposed to looking at it :)

There are now 17k packages for R on the CRAN sites.  With a bit of scripting (I should try to put 'more formally' together) we could probably serve a few of those besides mine.  It really is a very, very nice documentation system.

**Comment by squidfunk** [comment]
Thanks for the explanation! I can't make any promises, but if we can add styles that are reasonable for all users, it's definitely worth a try.

**Comment by eddelbuettel** [comment]
I found one last evening, and it no longer failed (and now I have difficulty remembering whether it failed ~ six months ago, or was simply 'typeset differently').  I since realized that the commonly-used package "does the same": by including markdown documentation source, the rendered page _of course_ reflects the overall setting for css etc (rather than was a a local markdown file may have head in its header).  For example, I use the minimal 'water' css wrapped via another R package [here in the standalone vignette](https://cran.r-project.org/web/packages/littler/vignettes/littler-examples.html) but via material-mkdocs it of course changes a little as [seen here](https://eddelbuettel.github.io/littler/vignettes/littler-examples/).  I think that is probably to be expected.

I will continue exploring, and will report back when I am stuck or otherwise confused :)
